src/NextGen/DonationForm/Blocks/DonationFormBlock/resources/app/templates/index.tsx

```diff
@@ -102,7 +110,12 @@ const template = {
 };
 
 // The following functions are used to retrieve the various templates for the form.
-function getTemplate<NodeProps>(type: string, section: string, htmlTag?: string): FC<NodeProps> {
+function getTemplate<NodeProps>(
+    type: string,
+    section: string,
+    htmlTag?: string,
+    template = activeFormDesignTemplate
```

Comment:
This is leftover from when I was playing around with bringing in the default `Goal` template so I don't have to re-write the html in classic form.  I ended up just importing it in directly.  I can see something like this being very useful though in practice.

Reply:
I'd rather not introduce something like this until we have a concrete use case for it.

Reply:
As a form design, If you're trying to get a template from the window - which one are you expecting? The getTemplate function is using the active template so if you're  overriding one of the template parts and also tried to get the template from the window it would cause a loop of inception.  So actually I think using the default template for the window makes the most sense.

Reply:
Hmm, I'm not quite following. Let's chat about this on Monday.

Reply:
The `getTemplate()` function is locating objects using the merge of the default (ours) and active form designs.  So here is an example of a loop:
 
**Scenario**
- classic design is active and supplies it's own `Goal` component
- inside it's `Goal` it uses `window.givewp.templates.getGoal()` and tries to render it (in this case classic doesn't need to change anything to the underlying structure - just add to it)
- this would keep trying to get `classic.layouts.goal` over and over instead of `default.layouts.goal` which is what it wants.  

```
export default function Goal(props: GoalProps) {
    const {
        targetAmountFormatted,
        totalRevenueFormatted,
        totalNumberOfDonationsOrDonors,
        totalNumberOfDonationsOrDonorsLabel,
    } = props;

    // this would now try to get itself inside of itself
   const GoalTemplate = window.givewp.templates.getGoal();

    return (
        <div style={{width: '100%'}}>
            <FormStats
                totalRevenue={totalRevenueFormatted}
                goalTargetAmount={targetAmountFormatted}
                totalNumberOfDonationsOrDonors={totalNumberOfDonationsOrDonors}
                totalNumberOfDonationsOrDonorsLabel={totalNumberOfDonationsOrDonorsLabel}
            />

           // LOOP HERE
            <GoalTemplate {...props} />
        </div>
    );
}
```

Reply:
Interesting! Makes sense. Sounds like we need a `getDefaultTemplate` function?

Reply:
Yeah, so what i'm thinking is 
- Internally we should always be getting the active template (merge of active and default design).  
- Externally (on the window / what we're giving to designs to consume) designs should always be getting the default core template we originally supplied.

Reply:
That makes sense. The only step we haven't added yet is add-ons, so it will be (in cascading order):
- Default templates
- Add-on tempaltes
- Form Design templates

Obviously the Form Design has access to its own templates, but it will need access to the default and add-on templates. Add-ons will only need access to default templates.

Reply:
Yup sounds good. We don't need to update that stuff in this PR but should in the near future.